brave/browser-laptop
# Add tools for generating a simulated Payment History for testing

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist. This fixes #4199 .

Adds npm script:  `npm run add-simulated-payment-history`

This generates a simulated Payment History, allowing testing of features that depend transactions  without spending BTC or waiting.

fixes #4199

Test Plan:
1. Clear user data (npm run clean userData)
2. Start Brave
3. Enable Brave Payments
4. Observe no transactions, empty wallet, Payment History link is **unavailable**
5. Run `npm run add-simulated-payment-history` 
6. Switch to Brave and Toggle Payments
7. Observe Payment History link is available
8. Open Payment History.
9. Observe 10 months of randomized Payments data
10. Open `/\<UserDataPath\>/ledger-state.json` and confirm transactions are present

![simulated_payments_history_script](https://cloud.githubusercontent.com/assets/6969859/19247116/7bf462a0-8ede-11e6-83d2-0725baf53742.gif)

cc @mrose17 , @diracdeltas
